common/graph/choose_test.go

```diff
@@ -24,3 +25,34 @@ func TestCombinationsExceed(t *testing.T) {
 	// N < K returns false
 	require.False(t, CombinationsExceed(20, 30, 0))
 }
+
+func TestChooseKoutOfN(t *testing.T) {
+	results := chooseKoutOfN(6, 4)
+	expectedSets := indiceSets{
+		&indiceSet{[]int{0, 1, 2, 3}},
+		&indiceSet{[]int{0, 1, 2, 4}},
+		&indiceSet{[]int{0, 1, 2, 5}},
+		&indiceSet{[]int{0, 1, 3, 4}},
+		&indiceSet{[]int{0, 1, 3, 5}},
+		&indiceSet{[]int{0, 1, 4, 5}},
+		&indiceSet{[]int{0, 2, 3, 4}},
+		&indiceSet{[]int{0, 2, 3, 5}},
+		&indiceSet{[]int{0, 2, 4, 5}},
+		&indiceSet{[]int{0, 3, 4, 5}},
+		&indiceSet{[]int{1, 2, 3, 4}},
+		&indiceSet{[]int{1, 2, 3, 5}},
+		&indiceSet{[]int{1, 2, 4, 5}},
+		&indiceSet{[]int{1, 3, 4, 5}},
+		&indiceSet{[]int{2, 3, 4, 5}},
+	}
+	for _, expected := range expectedSets {
```

Comment:
Since the function `chooseKoutOfN` is deterministic, we do not have to check for set equality but we can just convert the indiceSets to a string array:

```
func indiceSetsToStrings(sets indiceSets) []string {
	var res []string
	for _, set := range sets {
		res = append(res, fmt.Sprintf("%v", set.indices))
	}
	return res
}
```

and then it will be more friendly to read the test:

```
require.Equal(t, indiceSetsToStrings(expectedSets), indiceSetsToStrings(results))
```